Modules/_queuemodule.c

```diff
@@ -69,7 +210,10 @@ simplequeue_dealloc(simplequeueobject *self)
 static int
 simplequeue_traverse(simplequeueobject *self, visitproc visit, void *arg)
 {
-    Py_VISIT(self->lst);
+    RingBuf *buf = &self->buf;
+    for (Py_ssize_t i = 0, num_items = buf->num_items; i < num_items; i++) {
```

Comment:
The compiler will optimise the loop condition anyway, so I suggest writing this a little bit more idiomic:

```suggestion
    for (Py_ssize_t i = 0; i < buf->num_items; i++) {
```

... or:

```suggestion
    Py_ssize_t num_items = buf->num_items;
    for (Py_ssize_t i = 0; i < num_items; i++) {
```

Reply:
I don't think the compiler can optimize the loop condition here unless it can prove that the field is never mutated inside of the loop (which in this case it almost certainly cannot).

And the current form of the loop is preferable to the latter suggestion; it more tightly binds the scope of `num_items`.